### DISCUSSION TITLE
VipsOperation: class "svgload" not found

### DISCUSSION THREAD

**Discussion by IliyaKarkamov** [libvips/libvips#2208] [Q&A]
Hi,

I am trying to load SVG files using vips_svgload. But the function returns -1 and vips_error_buffer() returns the following message:
"VipsOperation: class "svgload" not found".

```cpp
VipsImage* pImage;

auto result = vips_svgload(fileName.c_str(), &pImage, dpi);

if (result < 0)
{
    auto s = vips_error_buffer();
    ...
}
```

I am initializing the library using VIPS_INIT as stated in the documentation. I was able to load images using vips_image_new_from_file including SVG files. But I want to set a different dpi when loading SVG files that is why I need the function vips_svgload.

Is this a known issue? Am I doing something wrong? Do I need to initialize the SVG loader somehow?

Best regards!

**Comment by jcupitt** [comment]
Hi @IliyaKarkamov,

It sounds like you've built libvips without SVG support. Try installing librsvg2-dev and reconfiguring and rebuilding libvips.

You can use `vips --vips-config` to verify that it's set up correctly. I see:

```
$ vips --vips-config | grep svg
SVG load with librsvg-2.0: yes
```

**Reply by IliyaKarkamov** [reply]
I am on Windows and I was looking for prebuilt 32-bit binaries. I found this thread [here](https://github.com/libvips/build-win32/issues/14#issuecomment-514530634) and downloaded the nip2 build and copied the vips binaries from there. 

I guess it is built without SVG support. I will try and build the library myself.

Thank you!

**Reply by jcupitt** [reply]
Ah OK. Just download one of the official releases:

https://github.com/libvips/libvips/releases/tag/v8.10.6

You need `vips-dev-w64-all-8.10.6.zip`.

**Reply by jcupitt** [reply]
You could try building yourself, but it's not easy on Windows, unfortunately. There's a thing to build it here:

https://github.com/libvips/build-win64-mxe

But it will still take a fair amount of work.

**Reply by IliyaKarkamov** [reply]
I need 32-bit binaries, not 64-bit. I suppose this is how to build them: https://github.com/libvips/build-win32

**Reply by jcupitt** [reply]
You can download a 32-bit binary here:

https://github.com/libvips/build-win64-mxe/releases/tag/v8.10.6

**Reply by IliyaKarkamov** [reply]
Thank you! I think that works.

But now when I call vips_svgload it returns the following error message:
```
svgload: no property named `ДР\x19'
svgload: no property named `РО\x19'
svgload: no property named `РО\x19'
```

**Comment by jcupitt** [comment]
That looks like some kind of unicode filename issue. Do you have a test file that shows the problem?

**Reply by IliyaKarkamov** [reply]
I think the problem was in the way I called the vips_svgload function:

```cpp
vips_svgload(fileName.c_str(), &pImage, dpi);
```

where it should be:

```cpp
vips_svgload(fileName.c_str(), &pImage, "dpi", dpi, NULL);
```

now it works without any error. 

Thank you!

**Reply by jcupitt** [reply]
Heh that would explain it. Yes, you need NULLs everywhere in C. The other APIs are a bit simpler.
